lang/stringx/stringx.go

```diff
@@ -0,0 +1,291 @@
+// note: size is unicode size.
+func PadLeftSpace(s string, size int) string {
+	return PadLeftChar(s, size, ' ')
+}
+
+// PadRightChar right pad a string with a specified character in a larger string(specified size).
+// if the size is less than the param string, the param string is returned.
+// note: size is unicode size.
+func PadRightChar(s string, size int, ch rune) string {
+	return padCharLeftOrRight(s, size, ch, false)
+}
+
+// PadLeftChar right pad a string with space character(' ') in a large string(specified size).
+// if the size is less than the param string, the param string is returned.
+// note: size is unicode size.
+func PadRightSpace(s string, size int) string {
+	return PadRightChar(s, size, ' ')
+}
+
+// PadCenterChar center pad a string with a specified character in a larger string(specified size).
+// if the size is less than the param string, the param string is returned.
+// note: size is unicode size.
+func PadCenterChar(s string, size int, ch rune) string {
+	if size <= 0 {
+		return s
+	}
+	length := Utf8Len(s)
+	pads := size - length
+	if pads <= 0 {
+		return s
+	}
+
+	// pad left
+	leftPads := pads / 2
+	if leftPads > 0 {
+		s = padRawLeftChar(s, ch, leftPads)
+	}
+	// pad right
+	rightPads := size - leftPads - length
+	if rightPads > 0 {
+		s = padRawRightChar(s, ch, rightPads)
+	}
+	return s
+}
+
+// PadCenterChar center pad a string with space character(' ') in a larger string(specified size).
+// if the size is less than the param string, the param string is returned.
+// note: size is unicode size.
+func PadCenterSpace(s string, size int) string {
+	return PadCenterChar(s, size, ' ')
+}
+
+func padCharLeftOrRight(s string, size int, ch rune, isLeft bool) string {
+	if size <= 0 {
+		return s
+	}
+	pads := size - Utf8Len(s)
+	if pads <= 0 {
+		return s
+	}
+	if isLeft {
+		return padRawLeftChar(s, ch, pads)
+	}
+	return padRawRightChar(s, ch, pads)
+}
+
+func padRawLeftChar(s string, ch rune, padSize int) string {
+	return RepeatChar(ch, padSize) + s
+}
+
+func padRawRightChar(s string, ch rune, padSize int) string {
+	return s + RepeatChar(ch, padSize)
+}
+
+// RepeatChar returns padding using the specified delimiter repeated to a given length.
+func RepeatChar(ch rune, repeat int) string {
+	if repeat <= 0 {
+		return ""
+	}
+	sb := strings.Builder{}
+	sb.Grow(repeat)
+	for i := 0; i < repeat; i++ {
+		sb.WriteRune(ch)
+	}
+	return sb.String()
+}
+
+// RemoveChar removes all occurrences of a specified character from the string.
+func RemoveChar(s string, rmVal rune) string {
+	if s == "" {
+		return s
+	}
+	sb := strings.Builder{}
+	sb.Grow(len(s) / 2)
+
+	for _, v := range s {
+		if v != rmVal {
+			sb.WriteRune(v)
+		}
+	}
+	return sb.String()
+}
+
+// RemoveString removes all occurrences of a substring from the string.
+func RemoveString(s, rmStr string) string {
+	if s == "" || rmStr == "" {
+		return s
+	}
+	return strings.ReplaceAll(s, rmStr, "")
```

Comment:
Feel like we could use `strings.ReplaceAll` directly, we don't need to wrap it into a function.

Reply:
> Feel like we could use `strings.ReplaceAll` directly, we don't need to wrap it into a function.

we provide a clear/breif wrapper is better than using strings.ReplaceAll directly.